### ISSUE TITLE
When openapi not contains format field for integer type, it is generated as Int32 C# equivalent type

### ISSUE THREAD

**Issue by vinaymadupathi** [christianhelle/refitter#167]
When openapi doesn't specify format for integer type, generated property is of Int32 whereas the value is in Int64 
range 

        "prop1": {
          "type": "integer"
        }

Refer Open API specs for Integer (https://swagger.io/docs/specification/data-models/data-types/#numbers)

**Comment by christianhelle** [comment]
@vinaymadupathi Thanks for taking the time to report this. I'll see what I can do. 

I will most likely not just enforce a `number` or `integer` default format because that would be a breaking change, but this could be specified as a command line argument instead. I'll look into what would be best

**Comment by christianhelle** [comment]
@all-contributors please add @vinaymadupathi for bugs

**Comment by kirides** [comment]
I'd argue that just using `"type": "integer"` is a bad openapi spec and should be fixed to include appropriate `"format": "int64" / "decimal" / "int32"` etc.

if this comes from an external source, it would be wise to tell the creator of the spec about the flaw in the openapi doc.

**Comment by christianhelle** [comment]
> I'd argue that just using `"type": "integer"` is a bad openapi spec and should be fixed to include appropriate `"format": "int64" / "decimal" / "int32"` etc.

I agree this is the correct course of action to take in this situation. 

> if this comes from an external source, it would be wise to tell the creator of the spec about the flaw in the openapi doc.

I've dealt with quite a lot of these bad OpenAPI specs myself, and when it comes to large corporate vendors, I noticed that informing them that there are errors in the spec doesn't necessarily mean that anything will be done about it. The worst case is that it remains as it is or gets fixed in a future release. You have probably tried this as well @kirides 

In these cases, I take a copy of the spec and adjust it locally. At the end of the day, HTTP API's use JSON, and there are only 3 primitive data types in JSON, so regardless of what numeric data type you use in C# it will always come out as a number

I'm guessing that the use case of @vinaymadupathi is to use tooling to skip adjusting the OpenAPI spec locally. Is this correct?

**Comment by Kebechet** [comment]
What is currently the right workaroud for this ? The doc file I use has:
```
purchased_at:
  description: The date when the purchase was made in ms since epoch
  nullable: false
  type: integer
  example: 1658399423658
```

And  openapi docs states: [data types CAN have an optional modifier property format](https://spec.openapis.org/oas/v3.1.0.html#data-types)
but when it is optional wouldnt be better to use C# `long` for type `integer` without `format` property ?

**Comment by HGCollier** [comment]
@Kebechet I agree, I think the default should be `long`, or at least provide an option to specify the default. Is this achievable @christianhelle?

**Comment by christianhelle** [comment]
It's been a while since I last checked this. 

Refitter piggy-backs on NSwag for parsing the OpenAPI spec and for generating the contracts. The only way I can think of to solve this is to add a option like `--integer-type` that accepts the type name and defaults to `int`. The implementation of this option will modify the parsed OpenAPI document and the integer type instances if a custom integer type is specified

This really just a workaround for scenarios where modifying the spec or requesting modification to the spec is not an option
